language/move-vm/types/src/values/values_impl.rs

```diff
@@ -140,6 +140,7 @@ pub enum IntegerValue {
 #[derive(Debug)]
 pub struct Struct {
     fields: Vec<ValueImpl>,
+    ty: Option<u64>,
```

Comment:
How about carrying the TypeTag here? and avoid the hash function call. this uses more memory but is better for performance.

Reply:
I'm not sure how much more perf we will be saving here as the resolving types in runtime to `TypeTag` is actually quite costly, and less memory efficient.